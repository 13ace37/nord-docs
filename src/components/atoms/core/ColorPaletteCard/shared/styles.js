/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides multiple styles for the `ColorPaletteCard` component.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.6.0
 */

import { rgba } from "polished";

import { colors, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

const floatingLineColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: rgba(colors.nord4, 0.8),
  [MODE_DARK_NIGHT_FROST]: rgba(colors.nord4, 0.5)
});

const dropShadowColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: rgba(colors.shadow.base[MODE_BRIGHT_SNOW_FLURRY], 0.35),
  [MODE_DARK_NIGHT_FROST]: rgba(colors.shadow.base[MODE_DARK_NIGHT_FROST], 0.2)
});

const fontColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord3,
  [MODE_DARK_NIGHT_FROST]: colors.nord4
});

export { dropShadowColor, floatingLineColor, fontColor };
