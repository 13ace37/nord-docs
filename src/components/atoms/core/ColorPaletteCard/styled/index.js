/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import Card from "./Card";
import CardWrapper from "./CardWrapper";
import Color from "./Color";
import ColorBox from "./ColorBox";
import FloatingTitle from "./FloatingTitle";
import Label from "./Label";
import Palette from "./Palette";
import Title from "./Title";

export { Card, CardWrapper, Color, ColorBox, FloatingTitle, Label, Palette, Title };
