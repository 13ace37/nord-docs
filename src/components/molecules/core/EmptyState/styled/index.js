/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import Illustration, { illustrationVariants } from "./Illustration";
import Headline from "./Headline";
import Subline from "./Subline";
import Wrapper from "./Wrapper";
import TextWrapper from "./TextWrapper";

export { Headline, Illustration, Subline, TextWrapper, Wrapper, illustrationVariants };
