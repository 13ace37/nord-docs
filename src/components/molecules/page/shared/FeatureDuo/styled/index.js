/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import Actions from "./Actions";
import Headline from "./Headline";
import Subline from "./Subline";
import Text from "./Text";
import Visualization from "./Visualization";
import Wrapper from "./Wrapper";

export { Actions, Headline, Subline, Text, Visualization, Wrapper };
