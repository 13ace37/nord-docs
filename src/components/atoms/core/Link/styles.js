/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides multiple styles for the `Link` component.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.5.0
 */

import { css } from "styled-components";
import { lighten, rgba } from "polished";

import { colors, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

const backgroundColorHover = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: rgba(colors.nord6, 0.45),
  [MODE_DARK_NIGHT_FROST]: rgba(colors.nord3, 0.8)
});

const fontColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord10,
  [MODE_DARK_NIGHT_FROST]: colors.nord8
});

const fontColorDecent = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord0,
  [MODE_DARK_NIGHT_FROST]: lighten(0.2, colors.font.base[MODE_DARK_NIGHT_FROST])
});

const fontColorHover = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord10
});

const fontColorHoverMinimal = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord0,
  [MODE_DARK_NIGHT_FROST]: lighten(0.2, colors.font.base[MODE_DARK_NIGHT_FROST])
});

const calmly = css`
  border-radius: 0.25em;
  color: ${fontColor};
  transition: color 0.2s ease-in-out, background-color 0.2s ease-in-out;

  &:hover,
  &:active,
  &:focus {
    background-color: ${backgroundColorHover};
    color: ${fontColorHover};
  }
`;

const decent = css`
  border-bottom: 1px solid ${fontColor};
  color: ${fontColorDecent};
  transition: color 0.2s ease-in-out;

  &:hover,
  &:active,
  &:focus {
    color: ${fontColor};
  }
`;

const minimal = css`
  border-bottom: 1px solid ${fontColor};
  transition: border-bottom-color 0.2s ease-in-out;

  &:hover,
  &:active,
  &:focus {
    border-bottom-color: ${fontColorHoverMinimal};
  }
`;

export { calmly, decent, minimal };
