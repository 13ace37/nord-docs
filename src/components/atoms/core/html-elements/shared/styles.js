/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides shared component styles.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.10.0
 */

import { colors, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

const borderColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.nord6,
  [MODE_DARK_NIGHT_FROST]: colors.nord2
});

/* eslint-disable-next-line import/prefer-default-export */
export { borderColor };
