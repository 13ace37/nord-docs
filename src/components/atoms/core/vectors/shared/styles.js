/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides shared vector graphic component styles.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */

import { css } from "styled-components";

import { colors, motion, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

const themeModeFillColor = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.font.base[MODE_BRIGHT_SNOW_FLURRY],
  [MODE_DARK_NIGHT_FROST]: colors.font.base[MODE_DARK_NIGHT_FROST]
});

/**
 * The value and transition for the fill color based on the active global theme mode.
 *
 * @since 0.3.0
 */
const themeModeFillColorStyles = css`
  fill: ${themeModeFillColor};
  transition: fill ${motion.speed.duration.transition.base.themeModeSwitch}s ease-in-out;
`;

/* eslint-disable-next-line import/prefer-default-export */
export { themeModeFillColorStyles };
