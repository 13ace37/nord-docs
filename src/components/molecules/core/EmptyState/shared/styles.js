/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides shared component styles.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */

import { css } from "styled-components";
import { darken } from "polished";

import { colors, motion, themedMode, MODE_BRIGHT_SNOW_FLURRY, MODE_DARK_NIGHT_FROST } from "styles/theme";

const svgFillColorOutlines = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.font.base[MODE_BRIGHT_SNOW_FLURRY],
  [MODE_DARK_NIGHT_FROST]: darken(0.045, colors.nord0)
});

const svgFillColorSnowflakes = themedMode({
  [MODE_BRIGHT_SNOW_FLURRY]: colors.font.base[MODE_BRIGHT_SNOW_FLURRY],
  [MODE_DARK_NIGHT_FROST]: colors.nord4
});

const svgTransitionFill = css`
  transition: fill ${motion.speed.duration.transition.base.themeModeSwitch}ms ease-in-out;
`;

export { svgFillColorOutlines, svgFillColorSnowflakes, svgTransitionFill };
