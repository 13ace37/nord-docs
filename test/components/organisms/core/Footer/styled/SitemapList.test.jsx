/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React from "react";
import { em } from "polished";

import { renderWithTheme } from "nord-docs-test-utils";
import { SitemapList } from "organisms/core/Footer/styled";
import { media } from "styles/theme";

describe("theme styles", () => {
  test("has 'start' text alignment for defined CSS breakpoints", () => {
    const mediaQuery = `(min-width: ${em(media.breakpoints.tabletLandscapeLowerBoundary)})`;
    const { container } = renderWithTheme(<SitemapList />);

    expect(container.firstChild).toHaveStyleRule("text-align", "start", {
      media: mediaQuery
    });
  });

  test("has centered text alignment", () => {
    const { container } = renderWithTheme(<SitemapList />);

    expect(container.firstChild).toHaveStyleRule("text-align", "center");
  });

  test("matches the snapshot", () => {
    const { container } = renderWithTheme(<SitemapList />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
