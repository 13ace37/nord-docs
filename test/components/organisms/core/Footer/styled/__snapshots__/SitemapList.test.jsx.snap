// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme styles matches the snapshot 1`] = `
.c0 {
  text-align: center;
}

@media (min-width:56.25em) {
  .c0 {
    text-align: start;
  }
}

<div
  class="c0"
/>
`;
